Add unused access key policy

ebattat opened the pull request:
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [ ] dependencies

## Description
<!--- Describe your changes below -->
Unused Access Key Policy:
Deactivate user access keys that are still active but have not been used in the last 90 days.
The threshold is defined by a constant parameter: UNUSED_ACCESS_KEY_DAYS = 90.

## For security reasons, all pull requests need to be approved first before running any automated CI

athiruma commented on cloud_governance/common/clouds/aws/iam/iam_operations.py:
```diff
@@ -158,3 +159,141 @@ def untag_role(self, role_name: str, tags: list):
+                return True
+        return False
+
+    def deactivate_user_access_keys(self, username):
+        """
+        Deactivates all active access keys for the given IAM user.
+
+        Args:
+            username (str): IAM user name
+        """
+        try:
+            access_keys = self.iam_client.list_access_keys(UserName=username)['AccessKeyMetadata']
+        except Exception as e:
+            logger.error(f"Failed to list access keys for user '{username}': {e}")
+            return
+
+        for idx, key in enumerate(access_keys, start=1):
+            label = f"Access key {idx}"
+            current_status = key['Status'].lower()
+            access_key_id = key['AccessKeyId']
+
+            if current_status == 'active':
+                try:
+                    self.iam_client.update_access_key(
+                        UserName=username,
+                        AccessKeyId=access_key_id,
+                        Status='Inactive'
+                    )
+                    logger.info(f"{label} deactivated for user '{username}'")
+                except Exception as e:
+                    logger.error(f"Failed to deactivate {label} for user '{username}': {e}")
+            else:
+                logger.info(f"{label} is already inactive for user '{username}'")
+
+        logger.info(f"Deactivate access keys for user '{username}' have been processed.")
```
We need to deactivate unused keys, not all keys. Her we should pass the inactive key_id to deactivate.

ebattat replied:
We deactivate it by resource_id=user_name.
You can find [here](https://github.com/redhat-performance/cloud-governance/blob/f38118882e88834c3e38307f73915b791b5104cf/cloud_governance/policy/helpers/aws/aws_policy_operations.py#L61-L62) the way we deactivate only specific user as other policies.
Pls let me know if it is ok.

athiruma replied:
Yeah, but if the user having multiple access_keys and one is recently used and one is used 90days back. We should delete the key that not using.

athiruma commented on cloud_governance/policy/aws/unused_access_key.py:
```diff
@@ -0,0 +1,56 @@
+        """
+        unused_access_keys = []
+        iam_users_access_keys = self._get_iam_users_access_keys()
+        for username, user_data in iam_users_access_keys.items():
+            last_activity_days = user_data['last_activity_days']
+            # Collect age_days only for active access keys
+            age_days_list = [
+                value[1] for key, value in user_data.items()
+                if key.startswith("Access key") and isinstance(value, list) and value[0] == "active"
+            ]
+            # "N/A"/None implies unused access key — fallback to UNUSED_ACCESS_KEY_MAX_DAY
+            age_days = min(age_days_list) if age_days_list else UNUSED_ACCESS_KEY_MAX_DAY
+            # if access key last_activity_days is "N/A", use age_days
+            if last_activity_days == "N/A":
+                last_activity_days = age_days
+            region = user_data['region']
+            user_name = username
```
```suggestion
        for user_name, user_data in iam_users_access_keys.items():
            last_activity_days = user_data['last_activity_days']
            # Collect age_days only for active access keys
            age_days_list = [
                value[1] for key, value in user_data.items()
                if key.startswith("Access key") and isinstance(value, list) and value[0] == "active"
            ]
            # "N/A"/None implies unused access key — fallback to UNUSED_ACCESS_KEY_MAX_DAY
            age_days = min(age_days_list) if age_days_list else UNUSED_ACCESS_KEY_MAX_DAY
            # if access key last_activity_days is "N/A", use age_days
            if last_activity_days == "N/A":
                last_activity_days = age_days
            region = user_data['region']
```

athiruma commented on cloud_governance/policy/helpers/aws/aws_policy_operations.py:
```diff
@@ -58,6 +58,8 @@ def _delete_resource(self, resource_id: str):
         try:
             if self._policy == 's3_inactive':
                 self._s3_client.delete_bucket(Bucket=resource_id)
+            elif self._policy == 'unused_access_key':
+                self._iam_operations.deactivate_user_access_keys(username=resource_id)
```
we need to pass the de-activate key-id.

athiruma commented:
Have you did the dry_run?

ebattat commented:
I run it locally and it works ok.
If we merge it, this policy will run as `dry_run=yes` in default, right ?

ebattat commented:
@athiruma,
I have updated the code based on your important comment.
Now, for each user, I have added an access_key_label to indicate "Access key 1" or "Access key 2."
By passing the access_key_label instead of the access_key_id, we keep our code more secure.
This means there are two checks per user, and each can trigger a different alert.
Additionally, we have added **kwargs to self._delete_resource(resource_id=resource_id, **kwargs) to allow passing more than one parameter—such as the user name and the access key label.
